src/terminal/adapter/adaptDispatch.cpp

```diff
@@ -2221,6 +2221,11 @@ bool AdaptDispatch::SetClipboard(const std::wstring_view content)
 bool AdaptDispatch::SetColorTableEntry(const size_t tableIndex, const DWORD dwColor)
 {
     _renderSettings.SetColorTableEntry(tableIndex, dwColor);
+    if (_renderSettings.GetRenderMode(RenderSettings::Mode::DistinguishableColors))
+    {
+        // Re-calculate the adjusted colors now that one of the entries has been changed
+        _renderSettings.MakeAdjustedColorArray();
+    }
```

Comment:
I think this needs to be called in `AdaptDispatch::AssignColor` too - at least for the `NormalText` variant - otherwise that's not going to work.

Reply:
Thank you for the catch!